src/lib/utils/stl_util.h

```diff
@@ -206,6 +218,17 @@ class BufferStuffer {
       std::span<uint8_t> m_buffer;
 };
 
+/**
+ * @brief Size guarded copy using spans.
+ *
+ * @param out output buffer
+ * @param in input buffer
+ */
+inline void copy_into(std::span<uint8_t> out, std::span<const uint8_t> in) {
+   BOTAN_ASSERT_NOMSG(in.size() == out.size());
+   std::copy(in.begin(), in.end(), out.begin());
+}
```

Comment:
We already have `copy_mem` in `mem_ops.h`. #3715 will introduce that for ranges as well.

Reply:
Yeah. Well, the current `copy_mem` only works on pointers by passing the size, but we wanted a function working on spans with an integrated size check. The implementation with ranges is exactly what we need. What do you suggest to do until then? I would keep `copy_into` for now and remove it after #3715 is merged.

Reply:
Sooner or later, we'll need to go through the code base and clean up these new range-based overloads once they are merged. No need to address here, I guess.